idaes/models_extra/power_generation/unit_models/heat_exchanger_common.py

```diff
@@ -0,0 +1,539 @@
+        doc="Total convective heat transfer coefficient on shell side",
+    )
+    def total_heat_transfer_coeff_shell(b, t, x):
+        # Retain in case we add back radiation
+        return b.conv_heat_transfer_coeff_shell[t, x]
+
+
+def scale_common(blk, shell, shell_has_pressure_change, make_reynolds, make_nusselt):
+    """Function to scale variables and constraints shared between
+    the CrossFlowHeatExchanger1D and Heater1D models.
+
+    Args:
+        blk : unit model on which components are being generated
+        shell: shell control volume
+        shell_units : derived units for property package of shell control volume
+        shell_has_pressure_change: bool about whether to make pressure change components
+        make_reynolds: bool about whether to create the Reynolds number
+        make_nusselt: bool about whether to create Nusselt number
+
+    Returns:
+        None
+    """
+
+    def gsf(obj):
+        return iscale.get_scaling_factor(obj, default=1, warning=True)
+
+    def ssf(obj, sf):
+        iscale.set_scaling_factor(obj, sf, overwrite=False)
+
+    def cst(con, sf):
+        iscale.constraint_scaling_transform(con, sf, overwrite=False)
+
+    sgsf = iscale.set_and_get_scaling_factor
+
+    sf_do_tube = iscale.get_scaling_factor(blk.do_tube, default=1 / value(blk.do_tube))
+    calculate_variable_from_constraint(
+        blk.area_flow_shell_min, blk.area_flow_shell_min_eqn
+    )
+    for t in blk.flowsheet().time:
+        for z in shell.length_domain:
+            sf_flow_mol_shell = gsf(shell.properties[t, z].flow_mol)
+
+            if make_reynolds:
+                # FIXME get better scaling later
+                ssf(blk.v_shell[t, z], 1 / 10)
+                cst(blk.v_shell_eqn[t, z], sf_flow_mol_shell)
+
+                # FIXME should get scaling of N_Re from defining eqn
+                sf_N_Re_shell = sgsf(blk.N_Re_shell[t, z], 1e-4)
+
+                sf_visc_d_shell = gsf(shell.properties[t, z].visc_d_phase["Vap"])
+                cst(blk.N_Re_shell_eqn[t, z], sf_N_Re_shell * sf_visc_d_shell)
+            if make_nusselt:
+                sf_k_shell = gsf(shell.properties[t, z].therm_cond_phase["Vap"])
+
+                sf_N_Nu_shell = sgsf(
+                    blk.N_Nu_shell[t, z], 1 / 0.33 * sf_N_Re_shell**0.6
+                )
+                cst(blk.N_Nu_shell_eqn[t, z], sf_N_Nu_shell)
+
+                sf_conv_heat_transfer_coeff_shell = sgsf(
+                    blk.conv_heat_transfer_coeff_shell[t, z],
+                    sf_N_Nu_shell * sf_k_shell / sf_do_tube,
+                )
+                cst(
+                    blk.conv_heat_transfer_coeff_shell_eqn[t, z],
+                    sf_conv_heat_transfer_coeff_shell * sf_do_tube,
+                )
+
+            # FIXME estimate from parameters
+            if blk.config.has_holdup:
+                s_U_holdup = gsf(blk.heat_holdup[t, z])
+                cst(blk.heat_holdup_eqn[t, z], s_U_holdup)
```

Comment:
Unit tests for scaling factors? Probably better to have some assertion about no numerical issues from the diagnostics toolbox.

Reply:
You really want both - the purpose of the unit test is not to check the scaling factors themselves but that the code runs smoothly (i.e. to help you narrow down where the issues is occurring rather than having to work throguh the entire unit-level method to find where it breaks). You do need a separate set of tests to make sure that the scaling factors do improve the condition of the model (for limited test cases at least).